### PR TITLE
Fix LazyMobXTable maxPage calculation

### PR DESCRIPTION
**Pull request by adamabeshouse**

### CODE REVIEW

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 1]
```diff
@@ -552,6 +549,8 @@ class LazyMobXTableStore<T> {
 
         this._page = 0;
         this.itemsPerPage = lazyMobXTableProps.initialItemsPerPage || 50;
+
+        reaction(()=>this.displayData.length, ()=>{ this.page = this.page; /* trigger recomputation of page when data length changes */});
```
Comment (alisman) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx]: this line seems potentially like a mistake
  this.page = this.page?

File: src/shared/components/lazyMobXTable/utils.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {SHOW_ALL_PAGE_SIZE} from "../paginationControls/PaginationControls";
+export function maxPage(displayDataLength:number, itemsPerPage:number) {
```
Comment (alisman) [inline src/shared/components/lazyMobXTable/utils.ts]: i think "pages" should not be zero indexed.  i.e. we don't ever have zero pages.  we have one page, two pages.

**Comment by inodb** [comment]
@adamabeshouse This might be better to `hotfix` no?

**Comment by alisman** [comment]
I agree.  So, switch this pr to hotfix/

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx (hunk 1 above)
Comment (adamabeshouse) [reply src/shared/components/lazyMobXTable/LazyMobXTable.tsx]: no its right - in `set page` it clamps the input value, and when the display data length changes it needs to be potentially clamped. So this just triggers that recomputation, as it says in the comment

File: src/shared/components/lazyMobXTable/utils.ts (hunk 2 above)
Comment (adamabeshouse) [reply src/shared/components/lazyMobXTable/utils.ts]: in my experience indexing math always works out easier when things are zero-indexed

**Comment by adamabeshouse** [comment]
see PR to hotfix https://github.com/cBioPortal/cbioportal-frontend/pull/611

**Pull request closed without merging**